src/annotate/celltypist/script.py

```diff
@@ -0,0 +1,105 @@
+import sys
+import logging
+import celltypist
+import mudata as mu
+import re
+import numpy as np
+
+## VIASH START
+par = {
+    "input": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_filtered_feature_bc_matrix_log_normalized.h5mu",
+    "output": "output.h5mu",
+    "modality": "rna",
+    "reference": "resources_test/annotation_test_data/TS_Blood_filtered.h5mu",
+    "model": None,
+    "reference_obs_targets": "cell_ontology_class",
+    "check_expression": False,
+    "feature_selection": True,
+    "majority_voting": True,
+    "output_compression": "gzip",
+    "var_query_gene_names": None,
+    "var_reference_gene_names": "ensemblid",
+    "reference_layer": None,
+    "output_obs_predictions": "celltypist_pred",
+    "output_obs_probabilities": "celltypist_probability",
+}
+meta = {
+}
+## VIASH END
+
+# START TEMPORARY WORKAROUND setup_logger
+def setup_logger():
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(sys.stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+
+def check_celltypist_format(indata):
+    if np.abs(np.expm1(indata[0]).sum()-10000) > 1:
+        return False
+    return True
+
+def main(par):
+    logger = setup_logger()
+
+    input_mudata = mu.read_h5mu(par["input"])
+    input_modality = input_mudata.mod[par["modality"]].copy()
+    
+    if par["var_query_gene_names"]:
+        input_modality.var.index = [re.sub("\\.[0-9]+$", "", s) for s in input_modality.var[par["var_query_gene_names"]]]
+
+    if par["model"]:
+        if par["reference"]:
+            logger.warning("Both 'model' and 'reference' are provided. 'model' will be used to make cell type predictions, 'reference' will be ignored.")
+        model = celltypist.models.Model.load(par["model"])
+        logger.info(f"Model features: {model.classifier.features}")
+        logger.info(input_modality.var)
+    
+    elif par["reference"]:
+        reference_modality = mu.read_h5mu(par["reference"]).mod[par["modality"]]
+                
+        if par["var_reference_gene_names"]:
+            reference_modality.var.index = [re.sub("\\.[0-9]+$", "", s) for s in reference_modality.var[par["var_reference_gene_names"]]]
+        
+        logger.info("Detecting common vars based on ensembl ids")
+        common_ens_ids = list(set(reference_modality.var.index).intersection(set(input_modality.var.index)))
+        
```

Comment:
There is an intersection for pandas indices:
https://pandas.pydata.org/docs/reference/api/pandas.Index.intersection.html